### ISSUE TITLE
PHP: google/protobuf requires bcmath extension

### ISSUE THREAD

**Issue by chingor13** [google/protobuf#4284]
The PHP library version requires the `bcmath` PHP extension but there is no documentation or declared dependency in `composer.json`.

If `bcmath` is not installed, you get:

`PHP Fatal error: Uncaught Error: Call to undefined function Google\Protobuf\Internal\bccomp() in /app/vendor/google/protobuf/php/src/Google/Protobuf/Internal/Message.php:882`

**Comment by atrauzzi** [comment]
Just hit this today as well, should this be moved from `suggest` to `requires`?

**Comment by TeBoring** [comment]
bcmath is only required for php implementation when json encode/decode is needed.

**Comment by TeBoring** [comment]
Probably, add a warning/error to explicitly name the error?

**Comment by atrauzzi** [comment]
That sounds good.  Yeah, if someone isn't using the JSON stuff, then it's not essential.

**Comment by rickslayer** [comment]
In my case I just install the module PHP bcmath

**Comment by atrauzzi** [comment]
Right, but if someone doesn't need it, then they're being forced to install an extension they don't plan on using.

**Comment by Eitol** [comment]
I am a user of the cloud vision library for PHP and this caused me a problem. They should fix it

**Comment by Eitol** [comment]
the problematic code:
<?php
require 'vendor/autoload.php';

use Google\Cloud\Vision\V1\Feature\Type;
use Google\Cloud\Vision\V1\ImageAnnotatorClient;
use Google\Cloud\Vision\V1\Likelihood;

$client = new ImageAnnotatorClient();

// Annotate an image, detecting faces.
$annotation = $client->annotateImage(
    fopen('/var/www/html/project/data/perfect.png', 'r'),
    [Type::IMAGE_PROPERTIES]
);

// Determine if the detected faces have headwear.
foreach ($annotation->getLocalizedObjectAnnotations() as $objectAnnotation) {
    $value = Likelihood::name($objectAnnotation->getHeadwearLikelihood());
    echo "Likelihood of headwear: $value" . PHP_EOL;
}

**Comment by paslandau** [comment]
FYI: If you can't install the extension for some reason (e.g. EOL legacy system and extension repository was removed), you can use https://github.com/phpseclib/bcmath_compat as a polyfill

**Comment by haberman** [comment]
Is any action required on this issue still?

**Comment by haberman** [comment]
Closing. There is another open issue for this here: https://github.com/protocolbuffers/protobuf/issues/4465
